php/Blockonomics.php

```diff
@@ -238,4 +292,267 @@ public function testSetup()
+        $this->load_blockonomics_template('nojs_payment_confirmation');
+    }
+
+    // Check if any pending payments linked to the order
+    public function is_payment_pending($order_id){
+        $orders = get_option('blockonomics_orders');
+        $network_confirmations = get_option("blockonomics_network_confirmation",2);
+        foreach ($orders[$order_id] as $addr => $order){
+            if ($order['status'] >= 0 && $order['status'] < $network_confirmations){
+                return true;
+            };
+        };
+        return false;
+    }
+
+    // Load the the checkout template in the page
+    public function load_checkout_template($order_id, $crypto){
+        // Check to send the user to nojs page
+        if($this->is_nojs_active()){
+            // Create or update the order for the nojs template
+            $this->process_order($order_id, $crypto);
+            $this->load_blockonomics_template('nojs_checkout');
+        }else{
+            $this->load_blockonomics_template('checkout');
+        }
+    }
+
+    // Load the the crypto options template in the page
+    public function load_crypto_options_template(){
+        // Check to send the user to nojs page
+        if($this->is_nojs_active()){
+            $this->load_blockonomics_template('nojs_crypto_options');
+        }
+        else{
+            $this->load_blockonomics_template('crypto_options');
+        }
+    }
+
+    // Redirect the user to the woocommerce finish order page
+    public function redirect_finish_order($order_id){
+        $wc_order = new WC_Order($order_id);
+        wp_redirect($wc_order->get_checkout_order_received_url());
+        exit();
+    }
+
+    // Check and update the bitcoin order
+    public function process_order($order_id, $crypto){
+        include_once 'Blockonomics.php';
+        $blockonomics = new Blockonomics;
+        global $woocommerce;
+        $orders = get_option('blockonomics_orders');
+        // Check post-meta for existing address by currency
+        $currentAddress = get_post_meta($order_id,$crypto .'_address', false);
+        if(! empty( $currentAddress ) && isset($currentAddress[0]) && $currentAddress[0] != "") {
+            $address = $currentAddress[0];
+        } else {
+            $responseObj = $blockonomics->new_address(get_option("blockonomics_callback_secret"), $crypto);
+            if($responseObj->response_code != 200) {
+                exit();
+            }
+            $address = $responseObj->address;
+        }
+        $wc_order = new WC_Order($order_id);
+        $order = $orders[$order_id][$address];
+        // Check if order has not expired yet
+        if ( isset($order['timestamp']) && $order['timestamp'] >= time() - get_option("blockonomics_timeperiod") * 60 ) {
+            $timestamp = $order['timestamp'];
+            $satoshi = $order['satoshi'];
+            $status = $order['status'];
+        }else{
+            // Refresh the timestamp and expected satoshi
```

Comment:
This also handles new orders, so the comment in misleading

Reply:
Fixed in 9e2864d